pkg/scheduler/framework/plugins/helper/node_affinity.go

```diff
@@ -72,7 +72,9 @@ func PodMatchesNodeSelectorAndAffinityTerms(pod *v1.Pod, node *v1.Node) bool {
 // nodeMatchesNodeSelectorTerms checks if a node's labels satisfy a list of node selector terms,
 // terms are ORed, and an empty list of terms will match nothing.
 func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
-	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, node.Labels, fields.Set{
-		"metadata.name": node.Name,
-	})
+	matches, err := v1helper.MatchNodeSelectorTerms(node, nodeSelectorTerms)
+	if err != nil {
+		klog.ErrorS(err, "Couldn't parse all NodeSelector terms", "terms", nodeSelectorTerms)
```

Comment:
same question about spewing errors on a single invalid selector evaluated against many nodes

Reply:
similar to my other comment, I think this is a scheduler-specific issue, so maybe this error should just be silenced in the plugin but caught instead in the PreFilter/PreScore stage by the scheduler, similar to how we did for podAffinity. wdyt @alculquicondor @ahg-g ?

Reply:
I'll take that as part of #95738

Reply:
Thanks Aldo, in that case we can ignore any error here because it should be caught earlier, correct?

Reply:
please add a `TODO(alculquicondor,#95738)`